noseiquela_orm/query.py

```diff
@@ -1,109 +1,117 @@
-from typing import Any, Dict, Tuple, Optional, Generator, TYPE_CHECKING
-from functools import partial
-from google.cloud.datastore.query import Query as GoogleQuery
+from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from .entity import Entity
+    from functools import partial
+    from typing import Any, Dict, Tuple, Optional, Generator, Iterable
+    from google.cloud.datastore.query import Query as GoogleQuery
+    from .entity import Model
 
 class QueryResult:
     def __init__(
         self,
-        query: GoogleQuery,
-        entity_instance: 'Entity',
-        limit: Optional[int]=None,
-        offset: Optional[int]=0,
-        retry: Optional[int]=None,
-        timeout: Optional[int]=None
-    ) -> None:
+        query: 'GoogleQuery',
+        entity_instance: 'Model',
+        limit: 'Optional[int]'=None,
+        offset: 'Optional[int]'=0,
+        retry: 'Optional[int]'=None,
+        timeout: 'Optional[int]'=None
+    ) -> 'None':
         self.query = query
         self.limit = limit
         self.offset = offset
         self.retry = retry
         self.timeout = timeout
         self.entity_instance = entity_instance
 
-    def first(self) -> Optional['Entity']:
-        result = list(self.query.fetch(
-            limit=1,
-            retry=self.retry,
-            timeout=self.timeout
-        ))
-        return self.entity_instance._create_from_google_entity(
-            result[0]
-        ) if result else None
-
-    def _get_query_result(self) -> Generator['Entity', None, None]:
-        for entity in self.query.fetch(
+    def _get_fetch_iterator(self):
+        return self.query.fetch(
             limit=self.limit,
             offset=self.offset,
             retry=self.retry,
             timeout=self.timeout
-        ):
-            yield self.entity_instance._create_from_google_entity(
+        )
+
+    def __iter__(self) -> 'Iterable[Model]':
+        for entity in self._get_fetch_iterator():
+            yield self.entity_instance._mount_from_google_entity(
                 entity
             )
 
-    def __iter__(self) -> Generator['Entity', None, None]:
-        return self._get_query_result()
-
-    def __next__(self) -> Generator['Entity', None, None]:
-        yield from self._get_query_result()
-
```

Comment:
It's a [`iterable`](https://docs.python.org/3/glossary.html#term-iterable) not a [`iterator`](https://docs.python.org/3/glossary.html#term-iterator).